### PR TITLE
kube-play: add support for HostPID

### LINKED ISSUE #17157: [Feature]: Support HostPID in podman kube play
### Feature request description

In k8s Pod, you can specify HostPID when container need access to Host PID (equivalent to --pid host). 

Currently, this is not supported by podman kube play, which is problematic when trying to migrate Docker compose yaml to k8s yaml usable with podman

### PR DESCRIPTION
**Pull request by danishprakash**
Adds support for hostPID in Pod manifest for podman kube play

Closes #17157 

#### Does this PR introduce a user-facing change?

```release-note
- play-kube: add support for hostPID in pod.spec
```
Signed-off-by: danishprakash <danish.prakash@suse.com>

Having some issues running the tests locally, hence the draft for now.

### CODE REVIEW

File: test/e2e/play_kube_test.go
Comment (vrothberg) [inline test/e2e/play_kube_test.go:4956]: We should also inspect the pod/container to make sure that the setting took the desired effect.

**Review by vrothberg — COMMENTED** [review]
Thanks for tackling the issue!  Can you squash the two commits into one?  I think it's preferable to have tests and changes in one commit (easier to bisect if needed).

File: pkg/specgen/generate/kube/kube.go
Comment (danishprakash) [inline pkg/specgen/generate/kube/kube.go:56]: @vrothberg do you think this check makes sense? and another log about telling the user that [it's an infra container but shares host pid](https://github.com/containers/podman/pull/17168/files#diff-cfef84d8077b70a186e602882259d5d3c6f8ecbf808b845e1f5a85c221df61e5R69)

Comment (vrothberg) [reply pkg/specgen/generate/kube/kube.go:56]: I do not feel strongly for or against adding a log.  If it would be helpful to you for debugging, it may be worth adding a debug log.

Comment (danishprakash) [reply pkg/specgen/generate/kube/kube.go:56]: Ok, I think I'll leave it at that then, it's a user-facing configuration option and makes sense for the user to decide how to use these.

**Comment by danishprakash** [comment]
@vrothberg since the change for `hostIPC` _seems_ similar, I was thinking of raising another PR for it, what do you suggest?

**Review by vrothberg — COMMENTED** [review]
LGTM, thank you.

**Comment by vrothberg** [comment]
> @vrothberg since the change for `hostIPC` _seems_ similar, I was thinking of raising another PR for it, what do you suggest?

That sounds good, thanks!

@containers/podman-maintainers PTAL

**Review by giuseppe — APPROVED** [review]
/lgtm
/hold

**Comment by vrothberg** [comment]
The test doesn't pass.

File: test/e2e/play_kube_test.go
Comment (Luap99) [inline test/e2e/play_kube_test.go:4961]:
```suggestion
		inspect := podmanTest.Podman([]string{"inspect", "test-hostpid-alpine", "--format", "{{ .HostConfig.PidMode }}"})
```
this will fix the test

File: test/e2e/play_kube_test.go
Comment (Luap99) [inline test/e2e/play_kube_test.go:4959]:
```suggestion
		Expect(logs.OutputToString()).To(Not(Equal("1")), "PID should never be 1 because of host pidns")
```
because  you have to repush anyway please add a description here so log readers can make sense of this error when it ever fails

Comment (danishprakash) [reply test/e2e/play_kube_test.go]: Right, that makes sense. Thanks for the suggestion

File: test/e2e/play_kube_test.go
Comment (danishprakash) [reply test/e2e/play_kube_test.go]: Ah, sorry about that. I referred to an existing test from here - https://github.com/danishprakash/podman/blob/29a90c3f8a1375b585dd3a870f7569ed2599426f/test/e2e/play_kube_test.go#L4249. Did I miss something?

File: pkg/domain/infra/abi/play.go
Comment (TomSweeneyRedHat) [inline pkg/domain/infra/abi/play.go:725]: Can we put this in alpha order here, i.e. move this line to 716.5?

**Comment by TomSweeneyRedHat** [comment]
One nit to keep my OCD in check, otherwise LGTM

File: pkg/domain/infra/abi/play.go
Comment (danishprakash) [reply pkg/domain/infra/abi/play.go:725]: Sorry, I completly forgot about that, fixing in https://github.com/containers/podman/pull/17186

**Pull request merged**
